src/routes/reservations/+page.svelte

```diff
@@ -1,46 +1,44 @@
 <script>
     // Framework imports
-    import { onMount } from "svelte";
     import { ProgressRadial } from "@skeletonlabs/skeleton";
+    import { goto } from "$app/navigation";
 
     // Import url routes and fetch method
     import { urls, tryFetchingRestricted } from "$lib/util.js";
+    import { loggedIn } from "$lib/stores/loggedIn.js";
     // Import custom reservation list component
     import ReservationList from "./reservationList.svelte";
-    import NotLoggedInComponent from "../notLoggedInComponent.svelte";
+    import Loading from "$lib/loading.svelte";
 
     // Definitions
-    export let data;
-
     let reservationData = new Promise((resolve, reject) => {});
 
-    // Functions
-    // Runs as soon as the component is mounted
-    onMount(async () => {
-        if (data.loggedIn) {
-            reservationData = new Promise(async (resolve, reject) => {
-                try {
-                    // Fetch backend for reservation Data with retry policy
-                    const response = await tryFetchingRestricted(
-                        urls.get.reservationPersonalData,
-                    );
-                    if (response.ok) {
-                        resolve(await response.json());
-                    } else {
-                        throw new Error("Error while fetching data");
-                    }
-                } catch (err) {
-                    reject(err); // Rethrow so Svelte can handle it
+    // Reactive statements
+    $: if ($loggedIn) {
+        reservationData = new Promise(async (resolve, reject) => {
+            try {
+                // Fetch backend for reservation Data with retry policy
+                const response = await tryFetchingRestricted(
+                    urls.get.reservationPersonalData,
+                );
+                if (response.ok) {
+                    resolve(await response.json());
+                } else {
+                    throw new Error("Error while fetching data");
                 }
-            });
-        }
-    });
+            } catch (err) {
+                reject(err); // Rethrow so Svelte can handle it
+            }
+        });
+    } else if ($loggedIn === false) {
```

Comment:
`if ($loggedIn) else if ($loggedIn === false)`... reicht da nicht auch nur `else`? Oder I guess `else if (!$loggedIn)` wenn `$loggedIn` mehr als nur `true` und `false` sein kann. Genau diese if-else-Struktur gibts noch ein paar mal mehr im Code, bitte wenn dann überall verändern.

Reply:
Ja gute Anmerkung. Der Code ist maybe nicht ganz clean an der Stelle. Der Store wird in `lib/stores/loggedIn.js` auf `null` initialisiert, da man nicht weiß, ob der user eingeloggt ist oder nicht. 
Das if statement muss daher prüfen, ob der Wert wirklich false ist. `!$loggedIn` würde dann natürlich auf falsey und nicht `false` prüfen und dann `null` auch zu falsey evaluieren. 

Mit Tyepsricpt würde man das obvisouly anders machen, mit JS ist mir kein besserer Ansatz eingefallen. Wenn du den hast, gerne her damit 👀

Reply:
Verstehe. Dann ist das sinnvoll so.